packages/blade/src/components/Input/DropdownInputTriggers/AutoComplete.tsx

```diff
@@ -0,0 +1,198 @@
+import BaseBox from '~components/Box/BaseBox';
+import { componentIds } from '~components/Dropdown/dropdownUtils';
+import { BladeElementRef } from '~utils/types';
+
+const useAutoCompleteHandlers = ({
+  props,
+  inputValue,
+  setInputValue,
+  getOptionValues,
+}: {
+  props: AutoCompleteProps;
+  setInputValue: (inputValue: string) => void;
+  inputValue: string;
+  getOptionValues: () => string[];
+}): {
+  onTriggerKeydown: BaseDropdownInputTriggerProps['onTriggerKeydown'];
+  onSelectionChange: BaseDropdownInputTriggerProps['onChange'];
+  onInputValueChange: BaseDropdownInputTriggerProps['onInputValueChange'];
+} => {
+  const {
+    onTriggerKeydown: onBaseDropdownInputKeydown,
+    isOpen,
+    setIsOpen,
+    selectedIndices,
+    setSelectedIndices,
+    setControlledValueIndices,
+    isControlled,
+    options,
+    setFilteredValues: setGlobalFilteredValues,
+    activeTagIndex,
+    setActiveTagIndex,
+    setActiveIndex,
+    filteredValues: globalFilteredValues,
+    selectionType,
+  } = useDropdown();
+
+  React.useEffect((): void => {
+    const firstItemOptionIndex = options.findIndex(
+      (option) => option.value === globalFilteredValues[0],
+    );
+
+    if (firstItemOptionIndex >= 0) {
+      setActiveIndex(firstItemOptionIndex);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [globalFilteredValues.length]);
+
+  const onInputValueChange: BaseDropdownInputTriggerProps['onInputValueChange'] = ({
+    name,
+    value,
+  }) => {
+    setInputValue(value ?? '');
+    props.onInputValueChange?.({ name, value });
+    setActiveTagIndex(-1);
+
+    if (!isOpen) {
+      setIsOpen(true);
+    }
+
+    if (!props.filteredValues) {
+      // eslint-disable-next-line no-lonely-if
+      if (value && options && options.length > 0) {
+        const filteredOptions = getOptionValues().filter((optionValue) =>
+          optionValue.toLowerCase().startsWith(value.toLowerCase()),
+        );
+        setGlobalFilteredValues(filteredOptions);
+      } else {
+        setGlobalFilteredValues(getOptionValues());
+      }
+    }
+  };
+
+  const onTriggerKeydown: BaseDropdownInputTriggerProps['onTriggerKeydown'] = (e) => {
+    if (e.key === 'Enter') {
+      setActiveTagIndex(-1);
+      setGlobalFilteredValues(getOptionValues());
+    } else if (
+      e.key === 'Backspace' &&
+      !inputValue &&
+      activeTagIndex < 0 &&
+      selectedIndices.length > 0
+    ) {
+      if (isControlled) {
+        setControlledValueIndices(selectedIndices.slice(0, -1));
+      } else {
+        setSelectedIndices(selectedIndices.slice(0, -1));
+      }
+    }
+    onBaseDropdownInputKeydown?.(e);
+  };
+
+  const onSelectionChange: BaseDropdownInputTriggerProps['onChange'] = ({ values }) => {
+    if (selectionType === 'multiple') {
+      setInputValue('');
+      props.onInputValueChange?.({ name: props.name, value: '' });
+      setActiveTagIndex(-1);
+      setGlobalFilteredValues(getOptionValues());
```

Comment:
What is this doing exactly? Its setting the filtered values to be the same as all options? Do we need to do that on ever selection change?

Reply:
yup when you select an item, you clear the input field so all options become visible for next selection. I'll add some comments